amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -466,15 +468,24 @@ private void recordsReceived(RecordFlow triggeringFlow, SubscribeToShardEvent re
                 return;
             }
 
-            List<KinesisClientRecord> records = recordBatchEvent.records().stream().map(KinesisClientRecord::fromRecord)
-                    .collect(Collectors.toList());
-            ProcessRecordsInput input = ProcessRecordsInput.builder().cacheEntryTime(Instant.now())
-                    .millisBehindLatest(recordBatchEvent.millisBehindLatest())
-                    .isAtShardEnd(recordBatchEvent.continuationSequenceNumber() == null).records(records).build();
-            FanoutRecordsRetrieved recordsRetrieved = new FanoutRecordsRetrieved(input,
-                    recordBatchEvent.continuationSequenceNumber(), triggeringFlow.subscribeToShardId);
-
             try {
+                if(!isValidEvent(recordBatchEvent)) {
+                    throw new RuntimeException("RecordBatchEvent for flow " + triggeringFlow.toString() + " is invalid."
```

Comment:
Can you add comments on what happens to the record processor flow when this exception is thrown?

Reply:
Will add comments